multi-tenancy.md

```diff
@@ -1,14 +1,57 @@
-## Multi Tenancy
+# Multi Tenancy
+
+This document describes how our framework integrations should support our multi-tenancy feature, with a sane set of defaults.  At the moment we are focusing on the subdomain-based used case.
+
+## Subdomain-Based Multi-Tenancy With Organizations
 
 The following design guidelines must be followed when adding multi-tenancy features to a framework integration:
 
-* On login, registration, and password reset, we should be able to dynamically resolve which account store should be used for the operation (all of these REST API operations accept an account store as an optional parameter, where the account store is identified by `nameKey` or `href`).  
+* This is an opt-in configuration, the developer must use `stormpath.web.multiTenancy=true` to declare that they want to opt-in to our default organization-based multi-tenant solution.
+
+* On login, registration, email verification, and password reset, we need to resolve which organization should be used for the operation (all of these REST API operations accept an account store as an optional parameter, where the account store is identified by `nameKey` or `href`).  
+
+* The resolution should be achieved with an "Organization Resolver".  The developer should be able to provide their own resolver, but our framework should provide a default resolver.  
+
+* The resolver type/interface should have the following characteristics:
+    - It accepts a HTTP request and response as parameters
```

Comment:
That seems a lot like the node standard to include request and response.  however, in PHP, it is not.  It is very odd to have request AND response passed in as parameters.